### PR TITLE
Simplify onboarding (VSC-426)

### PR DESCRIPTION
**Pull request by brianignacio5**
This branch is built to simplify user setup window.

* Cancellable downloads
* Express setup: No customization of ESP-IDF Tools and python virtual environment, just allow to choose ESP-IDF folder and install tools on default IDF_TOOLS_PATH `~/.espressif` or `%USERPROFILE%/.espressif`. If the user updates $IDF_TOOLS_PATH
* Custom setup: Allows the user to select the IDF_TOOLS_PATH or manually input each required tool and environment variable.
* Use default settings if existing ESP-IDF, ESP-IDF Tools and virtual environment are found. This will install `Debug Adapter and Extension requirements` in virtual environment found.

### CODE REVIEW

**Review by pwmb — CHANGES_REQUESTED** [review]
If I select the folder with IDF already there and try to install latest version of idf its going to this next page and UI freezes, ideally if folder had IDF already IDF, show the warning to user that this will override any changes they have there, and they can press `Yes` or `Continue` to proceed 

Screen Shot 2020-09-17 at 12 38 47 PM

---

`Use existing ESP-IDF setup` button is non functional!!

---

Old error is persisting, even after changing the directory (should be reactive)

Screen Shot 2020-09-17 at 12 42 43 PM

Same here

Screen Shot 2020-09-17 at 12 44 38 PM

**Comment by brianignacio5** [comment]
Hmm should redirected to install page if error happens. Will fix.

> `Use existing ESP-IDF setup` button is non functional!!

Any error in the logs or vscode Developer Tools console ?

**Comment by brianignacio5** [comment]
> 
> 
> If I select the folder with IDF already there and try to install latest version of idf its going to this next page and UI freezes, ideally if folder had IDF already IDF, show the warning to user that this will override any changes they have there, and they can press `Yes` or `Continue` to proceed
> 
> Screen Shot 2020-09-17 at 12 38 47 PM
> 
> `Use existing ESP-IDF setup` button is non functional!!
> 
> Old error is persisting, even after changing the directory (should be reactive)
> 
> Screen Shot 2020-09-17 at 12 42 43 PM
> 
> Same here
> 
> Screen Shot 2020-09-17 at 12 44 38 PM

This should be fixed. Please check the other errors.

File: src/views/setup/Home.vue [hunk 1]
```diff
@@ -0,0 +1,190 @@
+<template>
+  <div id="home">
+    <div class="centerize" v-if="!hasPrerequisites">
+      <p>
+        Before using this extension,
+        <a href="https://git-scm.com/downloads">Git</a>
```
Comment (pwmb) [inline src/views/setup/Home.vue:6]: Link on hover is not readable

Comment (pwmb) [reply src/views/setup/Home.vue:6]: I am getting no buttons to proceed or take action

Screen Shot 2020-10-14 at 12 07 43 PM

**Review by pwmb — CHANGES_REQUESTED** [review]

**Review by pwmb — CHANGES_REQUESTED** [review]
CSS Suggestions

- Add a box/border around so that these options just don't like text
- On mouse Hover on them change cursor to pointer to indicate link
- Also on hover change the text to blue (which is already the current behavior)

Screen Shot 2020-10-14 at 6 17 49 PM

- Left align the content in card (not center) and Right align the action buttons

Screen Shot 2020-10-14 at 6 23 10 PM


- First screen for both `Express` & `Advance` are identical is this intensional ?

**Comment by brianignacio5** [comment]
CSS suggestions should have been fixed.

> First screen for both Express & Advance are identical is this intensional ?

Yes because it is necessary for both advanced and express setup modes. Tools need ESP-IDF to be defined first in any case.

**Review by pwmb — CHANGES_REQUESTED** [review]
- `CSS`: On hover drop-down icon color needs to be updated
Screen Shot 2020-10-20 at 9 30 11 PM

- `CSS`: Those spinner icon are not spinning, also for ESP-IDF we have percentage then I think its better to show percentage and no spinner.

Screen Shot 2020-10-20 at 9 33 29 PM

- `CSS`: Error message is too in the way, also can't be canceled.

Screen Shot 2020-10-20 at 9 34 31 PM

- `Behaviour`: User should not be allowed to go to home or navigate away while download is happening, because it creates a race condition. (simple button disable should be fine)

Screen Shot 2020-10-20 at 9 34 04 PM


- `Question`: How user can set already downloaded IDF path, its should be part of advance mode I think?

**Comment by brianignacio5** [comment]
> Question: How user can set already downloaded IDF path, its should be part of advance mode I think?

In `Select ESP-IDF` dropdown options, there is an option to `Find ESP-IDF in your system`.

**Review by pwmb — APPROVED** [review]
LGTM

**Comment by kumekay** [comment]
I like the new download screen, it concise and informative. Great improvement.

My exprerience with the installer. Here are issues I encoutered on Windows 10 20h2 in WSL2 with ubuntu 20.04

What is the difference between advanced and express installation? Both of them lead to configuration page. 
image

Switching animation between screens is too long. It feels unresponsive. 0.4 s will be enough.

Missing space for label:
image

May the install create full path automatically, if it doesn't exist? Error messages doesn't fit.
image

The error is still on the next step, even after I created the path manually and installation goes normally:
image

I randomly receive 
image

Log doesn't contain anything useful:

```
xtensa-esp32-elf-gcc8_2_0-esp-2020r2-linux-amd64.tar.gz progress: 99.79% (83343.63 / 83515.85) KB
xtensa-esp32-elf-gcc8_2_0-esp-2020r2-linux-amd64.tar.gz progress: 99.81% (83359.63 / 83515.85) KB
xtensa-esp32-elf-gcc8_2_0-esp-2020r2-linux-amd64.tar.gz progress: 99.83% (83375.63 / 83515.85) KB
xtensa-esp32-elf-gcc8_2_0-esp-2020r2-linux-amd64.tar.gz progress: 99.85% (83391.63 / 83515.85) KB
xtensa-esp32-elf-gcc8_2_0-esp-2020r2-linux-amd64.tar.gz progress: 99.87% (83407.63 / 83515.85) KB
xtensa-esp32-elf-gcc8_2_0-esp-2020r2-linux-amd64.tar.gz progress: 99.89% (83423.63 / 83515.85) KB
xtensa-esp32-elf-gcc8_2_0-esp-2020r2-linux-amd64.tar.gz progress: 99.91% (83439.63 / 83515.85) KB
xtensa-esp32-elf-gcc8_2_0-esp-2020r2-linux-amd64.tar.gz progress: 99.93% (83455.63 / 83515.85) KB
xtensa-esp32-elf-gcc8_2_0-esp-2020r2-linux-amd64.tar.gz progress: 99.95% (83471.63 / 83515.85) KB
xtensa-esp32-elf-gcc8_2_0-esp-2020r2-linux-amd64.tar.gz progress: 99.97% (83487.63 / 83515.85) KB
xtensa-esp32-elf-gcc8_2_0-esp-2020r2-linux-amd64.tar.gz progress: 99.99% (83503.63 / 83515.85) KB
xtensa-esp32-elf-gcc8_2_0-esp-2020r2-linux-amd64.tar.gz progress: 100.00% (83515.85 / 83515.85) KB
```
Another issue that download progress is too verbose.

On download page progress bar for xtensa-esp32-elf  is filled with pervious attempt results. 
image

Wrong `\` slash for WSL path
image

**Comment by kumekay** [comment]
On Windows directly, install button doesn't work. Screenshot with error is attached:

image

**Comment by brianignacio5** [comment]
> What is the difference between advanced and express installation? Both of them lead to configuration page.

Advanced allows you to modify Tools directory folder or enter each tool path manually.

> With the second attempt it logged the error. Is codespace somehow hardcoded?

No it is not. `/home/codespace` is obtained from $HOME, the default tools directory is `$HOME/.espressif`.

Fixed other issues mentioned by @kumekay Please take a look again when possible.

**Review by pwmb — APPROVED** [review]

**Pull request merged**